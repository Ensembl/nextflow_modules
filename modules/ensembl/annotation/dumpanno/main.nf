// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process ANNOTATION_DUMPANNO {
    tag "${db.species}"
    label 'process_low'
    container "ensemblorg/ensembl-legacy-scripts:e112_APIv0.4"

    input:
        val(db)

    output:
        tuple val(db), val("functional_annotation"), path("*.json")

    when:
        task.ext.when == null || task.ext.when

    script:
        output = "${db.species}_functional_annotation.json"
        schema = "functional_annotation"
        password_arg = db.server.password ? "--pass ${db.server.password}" : ""
        """
        dump_functional_annotation.pl \
            --host $db.server.host \
            --port $db.server.port \
            --user $db.server.user \
            $password_arg \
            --dbname $db.server.database > $output

        schemas_json_validate --json_file $output --json_schema $schema

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        schemas_json_validate --version >> versions.yml
        """

    stub:
        """
        echo "No change, don't create a json file" > functional_annotation.json

        # Get version from genomio please
        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        cat <<-END_VERSIONS > versions.yml
        """
}
