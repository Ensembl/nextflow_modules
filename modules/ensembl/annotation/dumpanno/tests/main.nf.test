// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

nextflow_process {

    name "Test Process ANNOTATION_DUMPANNO"
    script "../main.nf"
    process "ANNOTATION_DUMPANNO"

    tag "annotation"
    tag "annotation/dumpanno"

       
    test("Dump functional annotation ") {
        when {

            options "-stub"

            process {
                """
                input[0] = [production_name:"test_db", server:[host:"test_host", port:"1111", user:"ensro"],species:"test_species"]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.functional_annotation.find { it[1] == "functional_annotation" } != null },
                { assert process.out.functional_annotation.flatten().find { it =~ /\.json$/ } != null },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
