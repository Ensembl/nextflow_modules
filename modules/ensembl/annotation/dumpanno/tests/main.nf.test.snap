{
    "Dump functional annotation ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "production_name": "test_db",
                            "server": {
                                "host": "test_host",
                                "port": "1111",
                                "user": "ensro"
                            },
                            "species": "test_species"
                        },
                        "functional_annotation",
                        "functional_annotation.json:md5,55d1982dcc9558c6290233151e103d13"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-10T21:32:05.999072669"
    }
}